picasso/src/main/java/com/squareup/picasso3/ContentStreamRequestHandler.java

```diff
@@ -60,4 +65,11 @@ Source getSource(Request request) throws FileNotFoundException {
     InputStream inputStream = contentResolver.openInputStream(request.uri);
     return inputStream == null ? null : Okio.source(inputStream);
   }
+
+  protected int getExifOrientation(Request request) throws IOException {
+    ContentResolver contentResolver = context.getContentResolver();
+    InputStream inputStream = contentResolver.openInputStream(request.uri);
+    ExifInterface exifInterface = new ExifInterface(inputStream);
```

Comment:
Any idea how much this reads from the stream? I wonder if it would be beneficial to do something similar to what we do for network requests and use the buffering input stream so that we can read twice. Provided this doesn't consume the whole stream, that is...

Reply:
Looks like it can consume the whole stream.  In ExifInterface#getJpegAttributes:
```            
 // EOI indicates the end of an image and in case of SOS, JPEG image stream starts and
 // the image data will terminate right after.
 if (marker == MARKER_EOI || marker == MARKER_SOS) {
     break;
 }
```